samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,224 @@
+    static final Duration TIMEOUT = Duration.ofSeconds(15);
+    static final Duration POLL_INTERVAL = Duration.ofMillis(1000);
+    static final String API_KEY_HEADER_KEY = "X-Api-Key";
+    static final String API_KEY_HEADER_VALUE = "password";
+    @RegisterExtension
+    static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            "provider",
+            Map.of(
+                    // Override 'edc.samples.04.asset.path' implicitly set via property 'edc.fs.config'.
+                    "edc.samples.04.asset.path", getFileFromRelativePath(SAMPLE_ASSET_FILE_PATH).getAbsolutePath(),
+                    "edc.fs.config", getFileFromRelativePath(PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
+            "consumer",
+            Map.of(
+                    "edc.fs.config", getFileFromRelativePath(CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    static final File DESTINATION_FILE = getFileFromRelativePath(FileTransferSampleTest.DESTINATION_FILE_PATH);
+    String contractNegotiationId;
+    String contractAgreementId;
+
+    /**
+     * Resolves a {@link File} instance from a relative path.
+     */
+    @NotNull
+    static File getFileFromRelativePath(String relativePath) {
+        return new File(TestUtils.findBuildRoot(), relativePath);
+    }
+
+    /**
+     * Run all sample steps in one single test.
+     * Note: Sample steps cannot be separated into single tests because {@link EdcRuntimeExtension}
+     * runs before each single test.
+     */
+    @Test
+    void runSampleSteps() throws IOException {
+        assertTestPrerequisites();
+
+        initiateContractNegotiation();
+        lookUpContractAgreementId();
+        requestTransferFile();
+        waitForDestinationFileExistence();
+
+        cleanTemporaryTestFiles();
+    }
+
+    /**
+     * Assert that prerequisites are fulfilled before running the test.
+     * This assertion checks only whether the file to be copied is not existing already.
+     */
+    void assertTestPrerequisites() {
+        assertThat(DESTINATION_FILE).doesNotExist();
+    }
+
+    /**
+     * Remove files created while running the tests.
+     * The copied file will be deleted.
+     */
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    void cleanTemporaryTestFiles() {
+        DESTINATION_FILE.delete();
+    }
+
+    /**
+     * Assert that the file to be copied exists at the expected location.
+     * This method waits a duration which is defined in {@link FileTransferSampleTest#TIMEOUT}.
+     */
+    void waitForDestinationFileExistence() {
+        await().atMost(TIMEOUT).pollInterval(POLL_INTERVAL).untilAsserted(() -> assertThat(DESTINATION_FILE).exists());
```

Comment:
```suggestion
        await().atMost(TIMEOUT).pollInterval(POLL_INTERVAL).untilAsserted(() -> assertThat(DESTINATION_FILE).hasSameBinaryContentAs(getFileFromRelativePath(SAMPLE_ASSET_FILE_PATH));
);
```

and adapt method name